Sources/FrontEnd/TypeChecking/TypeChecker.swift

```diff
@@ -1585,7 +1585,9 @@ public struct TypeChecker {
       : nil
   }
 
-  /// Returns a solution describing the best guess to type `subject` and its sub-expressions.
+  /// Returns the best solution satisfying `initialConstraints` and describing how to assign types
+  /// to `subject` and its sub-expressions and knowing it occurs in `scope` and is expected to have
+  /// a type compatible with `expectedType`.
```

Comment:
```suggestion
  /// Returns the best assignment of types
  /// to `subject` and its sub-expressions in `scope` satisfying `initialConstraints` and that is expected to have
  /// a type compatible with `expectedType`.
```

My suggestion improves on the wording a lot, but this "expected" thing is completely fuzzy.  Expected by whom?  What in the code causes such an "expectation?"  I'm guessing there's a better word.

Reply:
> Expected by whom? 

Expected by the caller? ;)

I have the feeling I'll commit 15 variants of this comment until it gets clear enough so I'll just explain how things work the best I can and shamelessly wait for your wisdom.

`expectedType` is used to propagate top-down type information to the sub-expressions of `subject`. For example, if we have `let x: Double = 42`, the checker will call `deduceType` for `42` with `Double` as the expected type.

Many different constructs may create expectations, not only type annotations. For example, if we have `f(x)`, it's possible the checker can expect a type `T` for `x` if it successfully resolved `f` as a function `(T) -> U`.

The comment says "compatible with" but perhaps it should say "subtype of".

Reply:
Then it's the _required_ type, or a supertype constraint.

Yes, "subtype" is more specific.

Reply:
Okay, I'll make a separate PR.